core/src/main/java/at/campus02/swd/game/strategy/AttackStrategy.java

```diff
@@ -1,45 +1,82 @@
 package at.campus02.swd.game.strategy;
 
-import at.campus02.swd.game.gameobjects.*;
+import at.campus02.swd.game.gameobjects.Enemy;
+import at.campus02.swd.game.gameobjects.UIPositionObserver;
 
 import java.util.Random;
-
+import java.util.Timer;
+import java.util.TimerTask;
 
 public class AttackStrategy implements MovementStrategy {
 
     private Enemy enemy;
-    private Player player;
+    private UIPositionObserver uiPositionObserver;
+    private Timer attackTimer;
 
-    public AttackStrategy(Enemy enemy, Player player) {
-        this.enemy = enemy;
-        this.player = player;
+    public AttackStrategy() {
+        attackTimer = new Timer();
     }
 
     @Override
-    public void execute() {
-    // TODO: direkte Position vom Player und Gegner umschreiben auf den Observer!
-        Random random = new Random();
-        float move = random.nextInt(20) + 1;
+    public void execute(Enemy enemy, UIPositionObserver uiPositionObserver) {
+        this.enemy = enemy;
+        this.uiPositionObserver = uiPositionObserver;
 
-            if (player.getPositionX() >enemy.getPositionX()){
-                enemy.setPosition(enemy.getPositionX()+move, enemy.getPositionY());
-            } else if (player.getPositionX() < enemy.getPositionX()) {
-                enemy.setPosition(enemy.getPositionX()-move, enemy.getPositionY());
-            }
-            else {
-                return;
+        attackTimer.scheduleAtFixedRate(new TimerTask() {
+            @Override
+            public void run() {
```

Comment:
Woher kommt da ein Override run() ? bzw. sollten wir sonst nicht einfach die Funktion performMovement() in run() umbenennen?

Reply:
Das habe ich mir auch gedacht aber komischerweise funktioniert das ganze dann nicht mehr. Ich komme leider nicht darauf warum...